src/bionics.cpp

```diff
@@ -2264,6 +2265,25 @@ void finalize_bionics()
     }
 }
 
+void bionic::set_flag( const std::string flag )
+{
+    if( !has_flag( flag ) ) {
```

Comment:
This check is redundant. Adding the flag when it's already there won't add it again (it can only exist once). Same for the removal: if the flag does not exist, it can't be removed and nothing will be done.